packages/tests/src/e2e/copilotplugin/CopilotPluginWithApiKeyForJs.tests.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @author Yimin Jin <yiminjin@microsoft.com>
+ */
+
+import { ProgrammingLanguage } from "@microsoft/teamsfx-core";
+import { replaceSecretKey, validateFiles } from "./helper";
+import { CopilotPluginCommonTest } from "./copilotPluginCommonTest";
+import * as path from "path";
+
+class CopilotPluginWithNoneAuthForJsCase extends CopilotPluginCommonTest {
+  public override async onAfterCreate(projectPath: string): Promise<void> {
+    const files: string[] = [
+      "appPackage/ai-plugin.json",
+      "appPackage/manifest.json",
+      "src/keyGen.js",
+    ];
+    await validateFiles(projectPath, files);
+
+    const userFile = path.resolve(projectPath, "env", `.env.dev.user`);
+    await replaceSecretKey(userFile);
+  }
+}
+
+new CopilotPluginWithNoneAuthForJsCase(
```

Comment:
We tried using a single file to create test cases for different languages, but we found it hard to find which language's template has a problem since the test cases in the same file were being executed asynchronously and out of order.

So now, we moved some common methods to the `CopilotPluginCommonTest` file and separated the different languages into their own files. This way, we can reduce the duplicate code and makes it easier to identify issues in specific language templates.

Reply:
We tried the one test case approach and it worked, the only problem was that the error messages were hard to distinguish, so we ended up splitting into multiple test cases.